ai8x_blocks.py

```diff
@@ -260,7 +259,9 @@ def __init__(self,
                                                kernel_size=1, batchnorm='Affine', bias=bias,
                                                eps=1e-03, momentum=0.01, **kwargs)
         # Skip connection
-        self.resid = ai8x.Add()
+        input_filters, output_filters = self.in_channels, self.out_channels
+        if self.stride == 1 and input_filters == output_filters:
+            self.resid = ai8x.Add()
 
```

Comment:
May it be better to add a warning/error message here for the else case?

Reply:
As the forward call has the same guard, I don't foresee a problem for the else case. I made this change to remove unused  ai8x.Add() modules from the network graph.

Reply:
I see, I was wondering if we would specifically want to prohibit users from defining models without any residual component for some reason like compatibility (should user be warned know if stride is not 1 and io filters do not match he cannot define ai8x compatible residual) In general i think these kind of warnings can be added to model definitions not to delay compatibility issues to synthesis time but it is totally fine as it is